app/components/settings/password-section.js

```diff
@@ -48,7 +48,21 @@ export default Component.extend(FormMixin, {
   actions: {
     submit() {
       this.onValid(() => {
-
+        let passwordData = this.getProperties('password_current', 'password_new');
+        let payload = {
+          'data': {
+            'old-password' : passwordData.password_current,
+            'new-password' : passwordData.password_new
+          }
+        };
+        this.get('loader')
+          .post('/auth/change-password', payload)
+          .then(() => {
+            this.get('notify').success(this.l10n.t('Password updated successfully'));
+          })
+          .catch(() => {
```

Comment:
If the server returns an error, display that error instead of saying `Unexpected error occurred`